Made content importer work as a job

sam-lord opened the pull request:
refs:  https://github.com/TryGhost/Toolbox/issues/431

Wrapped the current content importer function as a job, repurposing the members importer email to notify the user who initiates the content import when it is complete.

The API is changed a little to model after the members CSV import also - with a 201 accepted response if jobs are accepted, and a 202 if the job is processed inline and new data is created.



---

Main thing left on this PR before merging is that the db regression tests use the content import API, but would need to pass `forceInline` in order for it to have the same behaviour as before. I didn't want to expose this to the public API, but I think it would need to go there in order for those tests to work - unless we mocked out a property on the import manager 🤔

sam-lord committed:
Added a job for content imports

refs: https://github.com/TryGhost/Toolbox/issues/431

sam-lord committed:
Added tag to posts when importing content

refs: https://github.com/TryGhost/Toolbox/issues/431

sam-lord committed:
Remove bluebird from data-importer.js

no issue

Missed from previous commit

sam-lord committed:
Improved content import email

no issue

sam-lord committed:
Made content import API return semantic HTTP codes

no issue

naz commented on ghost/core/core/server/data/importer/import-manager.js:
```diff
@@ -351,45 +335,132 @@ class ImportManager {
+     * @returns {Promise<Object.<string, ImportResult>>}
      */
     async importFromFile(file, importOptions = {}) {
+        if (!importOptions.forceInline && !importOptions.runningInJob) {
```
I think instead of "forceInline" we could use the pattern of checking for test environment `env?.startsWith('testing')` (l[ike done here](https://github.com/TryGhost/Ghost/blob/9e35c4dd7a8c1904a5162bc064db2b3e04757ef5/ghost/core/core/server/services/members/service.js#L150)). No need to expose extra parameters in the API if it does not have a valid usecase.

sam-lord replied:
> I think instead of "forceInline" we could use the pattern of checking for test environment `env?.startsWith('testing')` (l[ike done here](https://github.com/TryGhost/Ghost/blob/9e35c4dd7a8c1904a5162bc064db2b3e04757ef5/ghost/core/core/server/services/members/service.js#L150)). No need to expose extra parameters in the API if it does not have a valid usecase.

Totally! I think I copied the `forceInline` out of the members importer but using env sounds perfect. Anything to get the tests passing :)

sam-lord committed:
Made content import run inline by default in test environment

refs: https://github.com/TryGhost/Toolbox/issues/431

sam-lord committed:
Updated importer test to use new object return format

no issue

sam-lord committed:
WIP importer modal

no issue

sam-lord committed:
Replaced "Open Importer" button with one that works

no issue

sam-lord committed:
Reverted unrelated changes in test file

no issue

peterzimon committed:
Restructured modal open logic

no refs.

- Navigating to `/settings/labs/import` opened the modal but also transitioned to a blank page in the background. Restructured content import modal so it's opening on top of the Labs page.

peterzimon committed:
Fixed file upload bug

peterzimon committed:
Fixed API params

sam-lord committed:
Fixed uploading zip files in import job

no issue
Zip files involved a call to `await`, which caused the microtask queue to be processed. The zip file is deleted during this time because the request has already completed.

sam-lord committed:
Prevent warnings from being displayed in email after content import

no issue

Warnings that aren't useful shouldn't be displayed, especially when this is likely someone's first experience of Ghost.

peterzimon committed:
Fixed content importer UI logic and email

- `result` parameter was wrongly passed to email
- email subject condition was reversed
- udpated email copy (still WIP)
- removed console.log from admin file selector

peterzimon committed:
Fixed condition in result email

sam-lord committed:
Fixed lint issues

no issue

rshbhgrg committed:
Updated ember action usage

rshbhgrg committed:
Revert "Updated ember action usage"

This reverts commit 31b4d34b903d8988a1f1d56c1c3bb45bcfa07e74.

rshbhgrg committed:
Added import modal lint errors to todo

sam-lord committed:
Fixed lint issue in email-template for content importer

sam-lord committed:
Merge branch 'main' into importer-job
